store/sqlstore/channel_store_categories.go

```diff
@@ -478,8 +476,11 @@ func (s SqlChannelStore) getSidebarCategoriesT(db dbSelecter, userId, teamId str
 	}
 
 	if _, err = db.Select(&categories, query, args...); err != nil {
+		return nil, errors.Wrapf(err, "failed to get categories for userId=%s,teamId=%s", userId, teamId)
```

Comment:
Ahh looks like you have to remove this.

Reply:
I think there's a bit of misusage of not founds errors in the store. Also I am not sure we are very consistent with the semantics. E.g., should an empty slice be considered a not found?

Reply:
Totally. Following the semantics of `database/sql`, for a query that returns a single row, a not found is meant to return an error. If the query returns multiple errors, no rows should return an empty slice.

But I am more for consistency than following the stdlib. We can decide whatever, but we should be consistent.